gluetool/glue.py

```diff
@@ -1557,160 +1629,176 @@ def has_module_class(item):
+        # type: (str, str) -> Any
         """
-        Load Python module from a file, if it contains ``gluetool`` modules. If the
-        file does not look like it contains ``gluetool`` modules, or when it's not
-        possible to import the Python module successfully, method simply warns
-        user and ignores the file.
+        If a file contains ``gluetool`` modules, import the file as a Python module. If the file does not look
+        like it contains ``gluetool`` modules, or when it's not possible to import the Python module successfully,
+        method simply warns user and ignores the file.
 
-        :param str import_name: name assigned to the imported module.
-        :param str filepath: path to a file.
+        :param str filepath: file to load.
+        :param str pm_name: Python module name for the loaded module.
         :returns: loaded Python module.
         :raises gluetool.glue.GlueError: when import failed.
         """
 
-        # Check content of the file, look for Glue and Module stuff
+        # Check content of the file, look for Glue and Module stuff.
         try:
-            if not self._check_module_file(filepath):
+            if not self._check_pm_file(filepath):
                 return
 
-        except GlueError as e:
-            self.warn("ignoring file '{}': {}".format(module_name, e.message))
+        except GlueError as exc:
+            self.warn("ignoring file '{}': {}".format(filepath, exc))
             return
 
-        # Try to import file as a Python module
-        import_name = 'gluetool.glue.{}-{}'.format(group, module_name)
-
+        # Try to import the file.
         try:
-            module = self._import_module(import_name, filepath)
+            pm = self._do_import_pm(filepath, pm_name)
 
-        except GlueError as e:
-            self.warn("ignoring module '{}': {}".format(module_name, e.message))
+        except GlueError as exc:
+            self.warn("ignoring file '{}': {}".format(filepath, exc))
             return
 
-        return module
-
-    def _load_gluetool_modules(self, group, module_name, filepath):
-        # type: (str, str, str) -> List[Tuple[str, Type[Module]]]
+        return pm
 
+    def _discover_gm_in_file(self, registry, filepath, pm_name, group_name):
+        # type: (Dict[str, DiscoveredModule], str, str, str) -> None
         """
-        Load ``gluetool`` modules from a file. Method attempts to import the file
-        as a Python module, and then checks its content and adds all `gluetool`
-        modules to internal module registry.
+        Discover ``gluetool`` modules in a file.
+
+        Attempts to import the file as a Python module, and then checks its content and looks for ``gluetool``
+        modules.
 
-        :param str group: module group.
-        :param str module_name: name assigned to the imported Python module.
+        :param dict(str, DiscoveredModule) registry: registry of modules to which new ones would be added.
```

Comment:
`ModuleRegistryType` ?

Reply:
Should be fixed.

Reply:
Will be fixed in a separate patch that should cover use of Mypy types in Sphinx directives in general.